presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -139,7 +142,18 @@ public void start(HiveSplitSource splitSource)
     {
         this.hiveSplitSource = splitSource;
         for (int i = 0; i < maxPartitionBatchSize; i++) {
-            ResumableTasks.submit(executor, new HiveSplitLoaderTask());
+            UserGroupInformation currentHdfsUser = getCurrentHdfsUser();
+            ResumableTasks.submit(executor, new HiveSplitLoaderTask(currentHdfsUser));
+        }
+    }
+
+    private UserGroupInformation getCurrentHdfsUser()
+    {
+        try {
+            return UserGroupInformation.getCurrentUser();
```

Comment:
Yeah - this section is called from top-level thread initiating split-loading process. And it is wrapped in `doAs` section by the caller.